Source/WebKit/Shared/libwpe/WebEventFactory.cpp

```diff
@@ -282,15 +284,24 @@ static WebKit::WebPlatformTouchPoint::TouchPointState stateForTouchPoint(int mai
 WebTouchEvent WebEventFactory::createWebTouchEvent(struct wpe_input_touch_event* event, float deviceScaleFactor)
 {
     auto type = WebEventType::NoType;
+    static NeverDestroyed<HashMap<int32_t, int32_t>> activeTrackingTouchPoints;
+    static int32_t uniqueTouchPointId = WebCore::mousePointerID + 1;
+    int32_t pointId;
+
     switch (event->type) {
     case wpe_input_touch_event_type_down:
         type = WebEventType::TouchStart;
+        activeTrackingTouchPoints.get().add(event->id, uniqueTouchPointId);
+        pointId = uniqueTouchPointId;
+        uniqueTouchPointId++;
         break;
     case wpe_input_touch_event_type_motion:
         type = WebEventType::TouchMove;
+        pointId = activeTrackingTouchPoints.get().get(event->id);
         break;
     case wpe_input_touch_event_type_up:
         type = WebEventType::TouchEnd;
+        pointId = activeTrackingTouchPoints.get().take(event->id);
         break;
```

Comment:
It looks like this only works if the down, motion, and up events all have the exact same `event->id`. So... that is actually true? Seems weird, but OK....

Reply:
I'll land this once you respond here.

Reply:
> It looks like this only works if the down, motion, and up events all have the exact same `event->id`. So... that is actually true? Seems weird, but OK....

It is true. `wpe_input_touch_event` and its id are based on libinput's seat slot of the touch event ([libinput_event_touch_get_seat_slot()](https://wayland.freedesktop.org/libinput/doc/latest/api/group__event__touch.html)).

The wayland protocol [describes this](https://gitlab.freedesktop.org/wayland/wayland/-/blob/1e259a255a07cd16b31fc99e6bba3e9368fae6d9/protocol/wayland.xml#L2536) as: "This touch point is assigned a unique ID. Future events from this touch point reference this ID. The ID ceases to be valid after a touch up event and may be reused in the future." Compositors implementing the wayland protocol usually just call `libinput_event_touch_get_seat_slot()` themselves.

So the event id is persistent for a touch point (meaning: down, motion [if any], up) for both the DRM platform and wayland.